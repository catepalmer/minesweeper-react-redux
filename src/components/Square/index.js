import React from 'react'
import styled from 'styled-components'

const StyledSquare = styled.div`
  border-color: hsla(0, 0%, 0%, 0.2);
  border-style: solid;
  border-width: 2px;
  color: gray;
  font-size: 4vh;
  font-weight: bold;
  line-height: 7vh;
  text-align: center;
  text-transform: uppercase;
`
StyledSquare.displayName = 'StyledSquare'

export default function Square ({ index, isMine }) {
  return (
    <StyledSquare index={index} isMine={isMine}>
      {isMine ? 'X' : 'O'}
    </StyledSquare>
  )
}