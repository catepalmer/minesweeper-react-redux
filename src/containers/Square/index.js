import { connect } from 'react-redux'

import Square from '../../components/Square'
import { blankSquareClicked, getMoves, squareClicked } from '../../state'
import { checkIfPlayed } from '../../utilities'

function mapStateToProps (state, { index }) {
  const moves = getMoves(state)

  return {
    isPlayed: checkIfPlayed(index, moves)
  }
}

function mapDispatchToProps (dispatch, { index }) {
  return {
    handleClick: () => dispatch(squareClicked(index)),
    handleClickBlank: () => dispatch(blankSquareClicked(index))
  }
}

export default connect(mapStateToProps, mapDispatchToProps)(Square)