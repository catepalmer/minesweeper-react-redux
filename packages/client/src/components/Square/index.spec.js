import React from 'react'
import { shallow } from 'enzyme'

import Square from '.'

describe('components:Square', () => {
  it('renders the Square with the proper cursor if a click handler is provided', () => {
    expect(
      toJson(shallow(<Square handleClick={() => null} index={0} isMine={false} minesTouching={1} />).dive())
    ).toMatchSnapshot()
  })

  it('renders the Square with the proper styles for a mine in the top left square', () => {
    expect(
      toJson(shallow(<Square isMine={true} index={0} />).dive())
    ).toMatchSnapshot()
  })

  it('renders the Square with the proper styles for a non-mine touching 4 mines in the top left square', () => {
    expect(
      toJson(shallow(<Square isMine={false} minesTouching={4} index={0} />).dive())
    ).toMatchSnapshot()
  })

  it('renders the Square with the proper styles for a mine in the top right square', () => {
    expect(
      toJson(shallow(<Square isMine={true} index={8} />).dive())
    ).toMatchSnapshot()
  })

  it('renders the Square with the proper styles for a non-mine touching 2 mines in the bottom left square', () => {
    expect(
      toJson(shallow(<Square isMine={false} minesTouching={2} index={72} />).dive())
    ).toMatchSnapshot()
  })

  it('renders the Square with the proper styles for a non-mine touching 0 mines in the bottom right square', () => {
    expect(
      toJson(shallow(<Square isMine={false} minesTouching={0} index={80} />).dive())
    ).toMatchSnapshot()
  })

})