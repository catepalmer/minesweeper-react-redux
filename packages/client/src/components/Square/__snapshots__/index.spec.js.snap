// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components:Square renders the Square with the proper cursor if a click handler is provided 1`] = `
.c0 {
  border-color: hsla(0,0%,0%,0.2);
  border-style: solid;
  border-width: 2px;
  cursor: pointer;
  font-size: 4vh;
  font-weight: bold;
  line-height: 7vh;
  text-align: center;
  text-transform: uppercase;
}

<div
  className="c0"
  onClick={[Function]}
/>
`;

exports[`components:Square renders the Square with the proper styles for a mine in the top left square 1`] = `
.c0 {
  border-color: hsla(0,0%,0%,0.2);
  border-style: solid;
  border-width: 2px;
  color: red;
  cursor: default;
  font-size: 4vh;
  font-weight: bold;
  line-height: 7vh;
  text-align: center;
  text-transform: uppercase;
}

<div
  className="c0"
>
  <styled.img
    src="mine.png"
  />
</div>
`;

exports[`components:Square renders the Square with the proper styles for a mine in the top right square 1`] = `
.c0 {
  border-color: hsla(0,0%,0%,0.2);
  border-style: solid;
  border-width: 2px;
  color: red;
  cursor: default;
  font-size: 4vh;
  font-weight: bold;
  line-height: 7vh;
  text-align: center;
  text-transform: uppercase;
}

<div
  className="c0"
>
  <styled.img
    src="mine.png"
  />
</div>
`;

exports[`components:Square renders the Square with the proper styles for a non-mine touching 0 mines in the bottom right square 1`] = `
.c0 {
  border-color: hsla(0,0%,0%,0.2);
  border-style: solid;
  border-width: 2px;
  cursor: default;
  font-size: 4vh;
  font-weight: bold;
  line-height: 7vh;
  text-align: center;
  text-transform: uppercase;
}

<div
  className="c0"
/>
`;

exports[`components:Square renders the Square with the proper styles for a non-mine touching 2 mines in the bottom left square 1`] = `
.c0 {
  border-color: hsla(0,0%,0%,0.2);
  border-style: solid;
  border-width: 2px;
  color: #307944;
  cursor: default;
  font-size: 4vh;
  font-weight: bold;
  line-height: 7vh;
  text-align: center;
  text-transform: uppercase;
}

<div
  className="c0"
>
  2
</div>
`;

exports[`components:Square renders the Square with the proper styles for a non-mine touching 4 mines in the top left square 1`] = `
.c0 {
  border-color: hsla(0,0%,0%,0.2);
  border-style: solid;
  border-width: 2px;
  color: #09114C;
  cursor: default;
  font-size: 4vh;
  font-weight: bold;
  line-height: 7vh;
  text-align: center;
  text-transform: uppercase;
}

<div
  className="c0"
>
  4
</div>
`;
